File: java/smallest-multiple.java
// Problem 5
// https://projecteuler.net/problem=5

import java.time.Duration;
import java.time.Instant;

class Problem5 {
    public static void main(String[] args) {
        Instant startTime = Instant.now();
        divisibility(20);
        System.out.println(Duration.between(startTime, Instant.now()).toString());
    }

    public static void divisibility(int limit) {
        int answer = 0;
        int number = 2520;
        int increment = 2520;
        while (answer == 0) {
            for (int i = 1; i < limit + 1; i++) {
                if(number % i != 0) {
                    break;
                }
                if(i == limit) {
                    answer = number;
                }
            }
            number += increment;
        }
        System.out.println(answer);
    }
}

///////////////////////
// Answer: 232792560 //
// Time: 0:00.004    //
///////////////////////

File: python/smallest-multiple.py
### Problem 51
### https://projecteuler.net/problem=5

from datetime import datetime
startTime = datetime.now()

divisor_max = 20        # find number divisible by all numbers from 1 to this variable
increment = 2520        
dividend = 2520         # set to same as increment

def divisible(test_number):
    for i in range(1, divisor_max+1):
        if test_number % i != 0:
            return False
    return True

while not divisible(dividend):
    dividend += increment
    divisible(dividend)

print(dividend)
print(datetime.now() - startTime)

# # # # # # # # # # #
# Answer: 232792560 #
# Time: 0:00.293    #
# # # # # # # # # # #